pkg/apis/core/validation/validation.go

```diff
@@ -2439,6 +2443,16 @@ func validateObjectFieldSelector(fs *core.ObjectFieldSelector, expressions *sets
 	return allErrs
 }
 
+func validateDownwardAPIHostIPs(fieldSel *core.ObjectFieldSelector, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if !opts.AllowHostIPsField {
```

Comment:
We should change:
1. (i need to confirm this on kep) dual stack req (your e2e test asserts that)
2. validate the correctness of IPs
3. we don't use gates in validation, it does not help with feat on/off scenarios

Reply:
This is to ensure that the original Pods that use this feature can still work normally after this feature is turned off.

Reply:
https://github.com/kubernetes/kubernetes/pull/101566/commits/405e9615d2e781edb46e9825088969c56a874e40

Reply:
In some cases, Enable this feature and create a pod using this feature. If this feature is disabled, status.hostIPs downwardAPI must be tolerated. When this feature is disabled, the pod is unable to use this downwardAPI.

related comment: https://github.com/kubernetes/kubernetes/pull/101566#discussion_r830937885

Reply:
``` yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  containers:
    - name: test-container
      image: gcr.io/google_containers/busybox
      command: [ "/bin/sh", "-c", "env" ]
      env:
        - name: MY_POD_HOST_IPS
          valueFrom:
            fieldRef:
              fieldPath: status.hostIPs
  restartPolicy: Never
```

That pod should be allowed despite the feature gate status

Reply:
That should be covered by `dropDisabledStatusFields` right?  If the gate is off and someone tries to set this field, it will be wiped out.  So you don't need to check the gate down here , which means you should not need this function at all, I think?

Reply:
Now we have a chance to revisit this :)